### DISCUSSION TITLE
Does Shadowsocks obfuscate the UDP traffic?

### DISCUSSION THREAD

**Discussion by mostafaznv** [shadowsocks/shadowsocks-rust#968] [Q&A]
Hi.
I'm using `shadowsocks-rust` using `v2ray-plugin` in `tls/websocket` mode.
Everything works fine but whenever I want to use **Telegram Voice Call**, it doesn't work

So, I have a question,
Does Shadowsocks(v2ray-plugin) obfuscate the UDP traffic?
Is there any advice to solve this problem? (server-side or client-side)

**Comment by database64128** [comment]
SIP003 plugins do not support UDP. Just use Shadowsocks without plugins.

**Reply by mostafaznv** [reply]
Thanks for reply.

Becuase of censorship in my country, I have to obfuscate the traffic.

without that, they drop all requests using their DPIs and firewalls.

Is there any workaround?

**Reply by database64128** [reply]
Does your censor block Shadowsocks UDP traffic? If your censor only targets Shadowsocks TCP traffic, you can configure 2 servers separately, one in TCP-only mode and another in UDP-only mode. The TCP server runs through your plugin. The UDP server runs directly.

**Reply by mostafaznv** [reply]
> Does your censor block Shadowsocks UDP traffic? If your censor only targets Shadowsocks TCP traffic, you can configure 2 servers separately, one in TCP-only mode and another in UDP-only mode. The TCP server runs through your plugin. The UDP server runs directly.

Thanks for this workaround. I'll try it.

I live in Iran and it is just like China (if it's not worse than that).
they blocked everything! everything possible protocol. and there are a few ways to bypass their censorship.

**Reply by database64128** [reply]
If it's just like China then you can just use Shadowsocks without plugins.

**Reply by mostafaznv** [reply]
> If it's just like China then you can just use Shadowsocks without plugins.

Is there any chance for them to find my ss server after a few days/weeks?

I know they use DPI techniques and they find vpn/proxy servers automatically.

**Reply by mostafaznv** [reply]
**Update:**
I configured a new server without any extra plugin. it didn't work without obfs at all 😕
It seems Iran is the worst place to live.

Is there any other solution?

**Reply by database64128** [reply]
If your country's firewall expects to see TLS Client Hello and Server Hello messages, and nothing more, you might be interested in [this experimental feature](https://github.com/database64128/shadowsocks-go#4-unsafe-stream-prefix) in my Shadowsocks implementation. You can have a pair of fake Client Hello and Server Hello messages as `unsafeRequestStreamPrefix` and `unsafeResponseStreamPrefix`, or anything else you like (e.g. [plain HTTP](https://github.com/shadowsocks/shadowsocks-org/issues/204#issuecomment-1266710067)).

Note that this feature is not in any stable release yet. You'll need to build it from the main branch or grab binaries from GitHub Actions.

**Reply by mostafaznv** [reply]
Many thanks ❤️

Another question (hopefully the last one 😄 ),
Is there any chance to support UDP in SIP003 plugins in future?

**Reply by database64128** [reply]
I see that you already upvoted shadowsocks/shadowsocks-org#180. 😄

Unfortunately, there does not seem to be much interest in this. I don't use SIP003 plugins myself. And I don't particularly like the way SIP003 interacts with the rest of the stack. As far as I know, the author of this project also doesn't use plugins, although he authored [a](https://github.com/zonyitoo/shadowsocks-yamux-plugin) [few](https://github.com/zonyitoo/sskcp-proxy) plugins (for fun I think).

Unless someone comes up with a reasonable spec, and really pushes to get everyone on board, I really don't see how this could happen.

**Reply by mostafaznv** [reply]
Oh... 😔
I hope you and all shadowsocks members change your mind about this.

Here in iran, censorship is at its worst level.

Instagram, Whatsapp, Telegram, YouTube, Twitter, Facebook, GitHub and everything else are banned by the regime and they use DPI techniques to drop almost all vpn protocols automatically.

Your shadowsocks proxy is not a simple proxy for us. it means our freedom ♥️

If I had more knowledge about network, I would help. but my knowledge is not enough for this awesome project 😕

**Reply by database64128** [reply]
> Note that this feature is not in any stable release yet. You'll need to build it from the main branch or grab binaries from GitHub Actions.

Update: A new release is out. Are you interested in testing whether this feature works for you?

**Reply by mostafaznv** [reply]
Yes. I'll try it later today.

**Reply by mostafaznv** [reply]
Hi.

I tried to test it.
It didn't work just like other shadowsocks implementations.

I don't know it's because of not using obfs plugin or it's because something else (Here in Iran, there are massive protests against the regime right now and internet is at its worst condition ever).

**Reply by database64128** [reply]
> I tried to test it.

Can you post your server and client configuration so I can take a look?

**Reply by dev4u** [reply]
@database64128 我觉得他需要的，是UDP的混淆协议(插件)
